atoma-service/src/handlers/embeddings.rs

```diff
@@ -137,19 +140,30 @@ pub async fn embeddings_handler(
             TOTAL_FAILED_TEXT_EMBEDDING_REQUESTS
                 .add(1, &[KeyValue::new("model", model.as_str().to_owned())]);
             TOTAL_FAILED_REQUESTS.add(1, &[KeyValue::new("model", model.as_str().to_owned())]);
-            let concurrent_requests = handle_concurrent_requests_count_decrement(
-                &state.concurrent_requests_per_stack,
-                stack_small_id,
-                "embeddings/embeddings_handler",
-            );
-            update_stack_num_compute_units(
-                &state.state_manager_sender,
-                stack_small_id,
-                estimated_total_compute_units,
-                0,
-                &endpoint,
-                concurrent_requests,
-            )?;
+            if let Some(stack_small_id) = stack_small_id {
+                let concurrent_requests = handle_concurrent_requests_count_decrement(
+                    &state.concurrent_requests_per_stack,
+                    stack_small_id,
+                    "embeddings/embeddings_handler",
+                );
+                update_stack_num_compute_units(
+                    &state.state_manager_sender,
+                    stack_small_id,
+                    estimated_total_compute_units,
+                    0,
+                    &endpoint,
+                    concurrent_requests,
+                )?;
+            } else {
+                update_fiat_amount(
```

Comment:
This calculation _could_ potentially lead to integer overflow if `estimated_total_compute_units` and `price_per_one_million_compute_units` are large enough integers. The integer overflow in release mode will wrap around, which could lead to incorrect financial calculations. Let's use `u128` during the calculations

```rust
// Rearranged calculation to prevent potential integer overflow
    // First divide to reduce the magnitude, then multiply
    let fiat_amount = (estimated_total_compute_units as u128 / ONE_MILLION as u128) 
        * price_per_one_million_compute_units as u128;
    
    update_fiat_amount(
        &state.state_manager_sender,
        user_address,
        fiat_amount,
        0,
        &endpoint,
    )?;
```

Reply:
This makes sense, we should address it everywhere in the code.

Reply:
I will change this, but just for fun, if the price for 1M tokens is $1000, and we do the same as usdc, 6 decimal places for the USD balance. It will overflow it there are estimation for more than 9B tokens, that's how big the i64 is :-)

Reply:
also I don't want to do the division first, because that will reduce the price to zero, if the token usage is e.g. 900k (which is not) then the equation above is zero, but I agree to put 128bit number just to be sure, I hope clippy doesn't complain about converting u128 back to i64